service/controller/resource/drainerinitializer/resource.go

```diff
@@ -157,8 +163,9 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 		for _, g := range o.AutoScalingGroups {
 			for _, i := range g.Instances {
 				c++
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking instance %s with state %s.", *i.InstanceId, *i.LifecycleState))
 
-				if *i.LifecycleState == autoscaling.LifecycleStateTerminatingWait {
+				if *i.LifecycleState == autoscaling.LifecycleStateTerminatingWait || *i.LifecycleState == autoscaling.LifecycleStateTerminatingProceed {
```

Comment:
when instance is already terminated but the lifecycle hook is still there the instnace is in state `LifecycleStateTerminatingProceed` and we should just finish the lifecycle hook to get rid of it